anonymization_PII_numbers.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# ***************************************************************************80
+#
+# python anonymization_PII_numbers.py -i <input file path>
+#
+# *****************************************************************************
+
+# standard imports
+import re
+import random
+
+# 3rd party imports
+import click
+import pandas
+
+@click.command()
+@click.option("-i","--input",required=True, help="input text file containing utterances")
+def main(input: str) -> None:
+    """Main Function"""
+
+    with open(input, mode="r", encoding="utf8") as f:
+        utterances = f.read()
+
+    utterances = utterances.split("\n")
+
+    df = pandas.DataFrame(data=utterances,columns=["utterances"])
+
+    print(f"Size of dataframe before processing: {df.shape}")
+
+    df["anonymized_phrases"] = ""
+
+    re_anydigit = re.compile(r"[0-9]+")
+    df["utterance_with_digits"] = df["utterances"].apply(lambda x: True if re.findall(re_anydigit,x) else False)
+
+    re_more_than_16_digit = re.compile(r"\b(?:\d[- ]?){16,}\d\b")
```

Comment:
There is a weird mix of boundary and no boundary regexes here.

For instance `\b(?:\d[- ]?){3}\d\b`
will be missing 1099g